src/Swarm/Game/Step.hs

```diff
@@ -584,10 +585,12 @@ tickRobot r = do
 --   runs it for one step, then calls itself recursively to continue
 --   stepping the robot.
 tickRobotRec :: (Has (State GameState) sig m, Has (Lift IO) sig m) => Robot -> m Robot
-tickRobotRec r
-  | isActive r && (r ^. runningAtomic || r ^. tickSteps > 0) =
-      stepRobot r >>= tickRobotRec
-  | otherwise = return r
+tickRobotRec r = do
+  time <- use ticks
+  if
+      | wantsToStep time r && (r ^. runningAtomic || r ^. tickSteps > 0) ->
+          stepRobot r >>= tickRobotRec
+      | otherwise -> return r
```

Comment:
Seems that `hlint` doesn't like this.

Reply:
Ugh, I hate `if ... then ... else ...` syntax in Haskell.  But I think I just remembered we already have a rule in `.hlint.yaml` to allow `case` expressions on Booleans, so I will switch to using that.